### ISSUE TITLE
albion online data project pushing data on different structure

### ISSUE THREAD

**Issue by GenoDuck** [Brannstroom/albiononline-refining-calculator#6]
https://imgur.com/a/k0LkUcM

albion online data project now pushes data in a different format, data should now be prefiltered so this don't happen again.

**Comment by GenoDuck** [comment]
I already have a solution for this, only made the solution for current prices, should be modified on the average prices pull function too

**Comment by GenoDuck** [comment]
Before:

`function pullCurrentPrices(resources) {
    console.log("pullCurrentPrices", resources);
    let offset = (resources ? 2 : 0);

    let table = document.getElementById("resouce-table-body");
    for(let i = 0; i < table.rows.length; i++) {
        table.rows[i].cells[4-offset].children[0].value = 0;
    }

    let flipped = document.getElementById("table-settings-flip-table").checked;
    if(flipped) {
        flipTable();
    }

    let element_id = resources ? "refining-city-resource" : "refining-city-product";

    let server = document.getElementById("server-select").value;
    let city = document.getElementById(element_id).value;

    let item_list = "";
    for(let i = 0; i < table.rows.length; i++) {
        let img_value = table.rows[i].cells[3-offset].innerHTML.split("/");
        let item_id = img_value[img_value.length-1].split(".")[0];

        if(item_id[item_id.length-1] >= '0' && item_id[item_id.length-1] <= '9') {
            let enchant = item_id[item_id.length-1];
            item_id = item_id.replace("_LEVEL"+enchant, "_LEVEL"+enchant+"@"+enchant);
        }

        item_list += item_id + ",";
    }

    let refining_resource = document.getElementById("refining-resource").value;
    let valArr = [0,1,2,2,2,2,3,3,3,3,4,4,4,4,5,5,5,5,6,6,6,6]

    let url = "https://"+server+".albion-online-data.com/api/v2/stats/prices/"+item_list+"?locations="+city;
    fetch(url)
    .then(response => response.json())
    .then(data => {
        for(let i = 0; i < table.rows.length; i++) {
            let index = refining_resource == "4" && !resources ? valArr[i] : i;
            let price;

            if (data[index] && data[index].sell_price_min !== null) {
                price = data[index].sell_price_min;
            }

            let price_cell = table.rows[i].cells[4-offset].children[0];
            price_cell.value = (price || 0);
           
            if(data[index]) {
                try {
                    changeAgeIndicator(price_cell, data[index].sell_price_min_date);
                } catch (error) {
                    console.log(error);
                }
            }
        }
    }
    ).then(() => {
        if(flipped) {
            flipTable();
        }
    }
    ).finally(() => {
        updateNumbers();
    });
}`


After:

`
function pullCurrentPrices(resources) {
    console.log("pullCurrentPrices", resources);
    let offset = (resources ? 2 : 0);

    let table = document.getElementById("resouce-table-body");
    for(let i = 0; i < table.rows.length; i++) {
        table.rows[i].cells[4-offset].children[0].value = 0;
    }

    let flipped = document.getElementById("table-settings-flip-table").checked;
    if(flipped) {
        flipTable();
    }

    let element_id = resources ? "refining-city-resource" : "refining-city-product";

    let server = document.getElementById("server-select").value;
    let city = document.getElementById(element_id).value;

    let item_list = "";
    for(let i = 0; i < table.rows.length; i++) {
        let img_value = table.rows[i].cells[3-offset].innerHTML.split("/");
        let item_id = img_value[img_value.length-1].split(".")[0];

        if(item_id[item_id.length-1] >= '0' && item_id[item_id.length-1] <= '9') {
            let enchant = item_id[item_id.length-1];
            item_id = item_id.replace("_LEVEL"+enchant, "_LEVEL"+enchant+"@"+enchant);
        }

        item_list += item_id + ",";
    }

    let refining_resource = document.getElementById("refining-resource").value;
    let valArr = [0, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 5, 5, 5, 5, 6, 6, 6, 6];

    let url = "https://" + server + ".albion-online-data.com/api/v2/stats/prices/" + item_list + "?locations=" + city;
    fetch(url)
        .then(response => response.json())
        .then(data => {
            // Filters duplicate data, keeping only the first occurrence
            let seenItems = new Set();
            let filteredData = data.filter(item => {
                if (!seenItems.has(item.item_id)) {
                    seenItems.add(item.item_id);
                    return true;
                }
                return false;
            });

            for (let i = 0; i < table.rows.length; i++) {
                let index = refining_resource == "4" ? valArr[i] : i;
                let price;

                if (filteredData[index] && filteredData[index].sell_price_min !== null) {
                    price = filteredData[index].sell_price_min;
                }

                let price_cell = table.rows[i].cells[4 - offset].children[0];

                price_cell.value = (price || 0);

                if (filteredData[index]) {
                    changeAgeIndicator(price_cell, filteredData[index].sell_price_min_date);
                }
            }
        })
        .then(() => {
            if (flipped) {
                flipTable();
            }
        })
        .finally(() => {
            updateNumbers();
        });

}
`

**Comment by GenoDuck** [comment]
First time doing this on github, dont know how to commit on a public project haha

With these changes it should be solved

https://imgur.com/a/6RHTkh8

**Comment by Brannstroom** [comment]
Thank you for creating an issue on the problem.

As you mentioned, AOD made changes which adds a quality attribute to resources, while resources are all normal (quality 1). I solved this in 6e6195cf503805d4bb205bfc94cddf8184e837cc by adding `qualitites=1` as a parameter to the GET request to AOD. This might not be the best solution, but it's the solution I found that requires the least amount of data processing.
